terraform/03_gke_cluster.tf

```diff
@@ -135,20 +135,28 @@ resource "null_resource" "set_gke_context" {
   ]
 }
 
+# Install Istio into the GKE cluster
+resource "null_resource" "install_istio" {
+  provisioner "local-exec" {
+    command = "./istio/install_istio.sh"
+  }
+
+  depends_on = ["null_resource.set_gke_context"]
+}
+
 # Deploy microservices into GKE cluster 
 resource "null_resource" "deploy_services" {
   provisioner "local-exec" {
     command = "kubectl apply -f ../kubernetes-manifests"
   }
 
-  depends_on = ["null_resource.set_gke_context"]
+  depends_on = ["null_resource.install_istio"]
 }
 
-# There is no reliable way to do deployment verification with kubernetes
-# For the purposes of Sandbox, we can mitigate by waiting a few sec to ensure kubectl apply completes
+# We wait for all of our microservices to become available on kubernetes
 resource "null_resource" "delay" {
   provisioner "local-exec" {
-    command = "sleep 5"
+    command = "kubectl wait --for=condition=available --timeout=600s deployment/adservice && kubectl wait --for=condition=available --timeout=600s deployment/cartservice && kubectl wait --for=condition=available --timeout=600s deployment/checkoutservice && kubectl wait --for=condition=available --timeout=600s deployment/currencyservice && kubectl wait --for=condition=available --timeout=600s deployment/emailservice && kubectl wait --for=condition=available --timeout=600s deployment/frontend && kubectl wait --for=condition=available --timeout=600s deployment/paymentservice && kubectl wait --for=condition=available --timeout=600s deployment/productcatalogservice && kubectl wait --for=condition=available --timeout=600s deployment/recommendationservice"
```

Comment:
Can this be split into multiple lines for clarity?

Reply:
@nicklin37 I just checked and this is also valid to cut down on the size:

`kubectl wait deployment --for=condition=available --timeout=600s adservice cartservice checkoutservice currencyservice emailservice frontend paymentservice productcatalogservice recommendationservice shippingservice`

That's still pretty long though, so breaking it up would be ideal if that's possible

Reply:
Thanks Terraform really only likes single lines for quoted strings so I'm using an EOT  'heredoc' style. It causes added delimiters but it breaks it up. Let me know if that's better.